geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/BinAggregatingIterator.scala

```diff
@@ -114,8 +103,15 @@ object BinAggregatingIterator extends LazyLogging {
         val id = index.getIdFromRow(row.getBytes, 0, row.getLength, deserialized)
         deserialized.getIdentifier.asInstanceOf[FeatureIdImpl].setID(id)
         val values = Array[AnyRef](encoder.encode(deserialized), GeometryUtils.zeroPoint)
-        new ScalaSimpleFeature(BinaryOutputEncoder.BinEncodedSft, deserialized.getID, values)
+        new ScalaSimpleFeature(BinEncodedSft, deserialized.getID, values)
       }
     }
   }
+
+  /**
+    * Adapts the iterator to create simple features
+    */
+  class AccumuloBinResultsToFeatures extends BinResultsToFeatures[Entry[Key, Value]] {
+    override protected def bytes(result: Entry[Key, Value]): Array[Byte] = result.getValue.get()
```

Comment:
As an over-refactoring, it seems like you'd be able to have a trait for Accumulo to mix-in this def;).

Reply:
haha, seems like overkill for one line :)